File: JavaFundamentals/Course2_JavaSyntax/Homework/src/CharacterMultiplier.java
import java.util.Scanner;

public class CharacterMultiplier {
    public static int output (String firstString, String secondString){
        int sum = 0;
        String maxString = "";

        int maxLength = Math.max(firstString.length(), secondString.length());
        int minLength = Math.min(firstString.length(), secondString.length());

        if (firstString.length() >= secondString.length()) {
            maxString = firstString;
        }
        else {
            maxString = secondString;
        }
        for (int i = 0; i < maxLength; i++) {
            if (i < minLength){
                sum += firstString.charAt(i) * secondString.charAt(i);
            }
            else {
                sum += maxString.charAt(i);
            }
        }
        return sum;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str1 = sc.next();
        String str2 = sc.next();

        System.out.println(output(str1, str2));
    }
}

File: AdvancedC#/5ManualStringProcessing/StringProcessingExercise/12CharacterMultiplier/CharacterMultiplier.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class CharacterMultiplier
{
    public static void Main()
    {
        string[] elements = Console.ReadLine().Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
        string str1 = elements[0];
        string str2 = elements[1];

        int sum = 0;
        string maxString = string.Empty;

        int maxLength = Math.Max(str1.Length, str2.Length);
        int minLength = Math.Min(str1.Length, str2.Length);

        if (str1.Length >= str2.Length)
        {
            maxString = str1;
        }
        else
        {
            maxString = str2;
        }
        for (int i = 0; i < maxLength; i++)
        {
            if (i < minLength)
            {
                sum += str1[i] * str2[i];
            }
            else
            {
                sum += maxString[i];
            }
        }
        Console.WriteLine(sum);
    }
}
